components/kyma-environment-broker/internal/process/upgrade_kyma/cls_audit_log.go

```diff
@@ -0,0 +1,83 @@
+package upgrade_kyma
+
+import (
+	"fmt"
+	"strings"
+	"time"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/cls"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/auditlog"
+
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/process"
+	"github.com/kyma-project/control-plane/components/kyma-environment-broker/internal/storage"
+	"github.com/kyma-project/control-plane/components/provisioner/pkg/gqlschema"
+	"github.com/sirupsen/logrus"
+	"github.com/spf13/afero"
+)
+
+type ClsUpgradeAuditLogOverridesStep struct {
+	operationManager *process.UpgradeKymaOperationManager
+	fs               afero.Fs
+	auditLogConfig   auditlog.Config
+	secretKey        string
+}
+
+func (alo *ClsUpgradeAuditLogOverridesStep) Name() string {
+	return "CLS_Audit_Log_Overrides"
+}
+
+func NewClsUpgradeAuditLogOverridesStep(os storage.Operations, cfg auditlog.Config, secretKey string) *ClsUpgradeAuditLogOverridesStep {
+	fileSystem := afero.NewOsFs()
+
+	return &ClsUpgradeAuditLogOverridesStep{
+		process.NewUpgradeKymaOperationManager(os),
+		fileSystem,
+		cfg,
+		secretKey,
+	}
+}
+
+func (alo *ClsUpgradeAuditLogOverridesStep) Run(operation internal.UpgradeKymaOperation, logger logrus.FieldLogger) (internal.UpgradeKymaOperation, time.Duration, error) {
+	luaScript, err := afero.ReadFile(alo.fs, "/auditlog-script/script")
+	if err != nil {
+		logger.Errorf("Unable to read audit config script: %v", err)
+		return operation, 0, err
+	}
+
+	replaceSubAccountID := strings.Replace(string(luaScript), "sub_account_id", operation.ProvisioningParameters.ErsContext.SubAccountID, -1)
+	replaceTenantID := strings.Replace(replaceSubAccountID, "tenant_id", alo.auditLogConfig.Tenant, -1)
+
+	auditlogOverrideParams, err := auditlog.PrepareOverrideParams(&alo.auditLogConfig, alo.secretKey, operation.Cls.Overrides)
+	if err != nil {
+		logger.Errorf("Unable to prepare Audit Log override parameters: %v", err)
+		return operation, 0, err
```

Comment:
please make the operation failed (and also in other cases): return s.operationManager.OperationFailed(operation, msg)